dialog.go

```diff
@@ -0,0 +1,104 @@
+	Type             string                 `json:"type"`                           //Required. Allowed values: "text", "textarea", "select".
+	Placeholder      string                 `json:"placeholder,omitempty"`          //Optional.
+	Optional         bool                   `json:"optional,omitempty"`             //Optional. Default value is false
+	Value            string                 `json:"value,omitempty"`                //Optional.
+	DataSource       string                 `json:"data_source,omitempty"`          //Optional. Allowed values: "users", "channels", "conversations", "external".
+	SelectedOptions  string                 `json:"selected_options,omitempty"`     //Optional. Default value for "external" only
+	Options          []DialogElementOption  `json:"options,omitempty"`              //One of options or option_groups is required.
+	OptionGroups     []DialogElementOption  `json:"option_groups,omitempty"`        //Provide up to 100 options.
+}
+
+type DialogElementOption struct {
+	Label  string   `json:"label"`	//Required.
+	Value  string   `json:"value"`	//Required.
+}
+
+// DialogCallback is sent from Slack when a user submits a form from within a dialog
+type DialogCallback struct {
+	Type         string             `json:"type"`
+	CallbackID   string             `json:"callback_id"`
+	Team         Team               `json:"team"`
+	Channel      Channel            `json:"channel"`
+	User         User               `json:"user"`
+	ActionTs     string             `json:"action_ts"`
+	Token        string             `json:"token"`
+	ResponseURL  string             `json:"response_url"`
+	Submission   map[string]string  `json:"submission"`
+}
+
+// DialogSuggestionCallback is sent from Slack when a user types in a select field with an external data source
+type DialogSuggestionCallback struct {
+	Type         string             `json:"type"`
+	Token        string             `json:"token"`
+	ActionTs     string             `json:"action_ts"`
+	Team         Team               `json:"team"`
+	User         User               `json:"user"`
+	Channel      Channel            `json:"channel"`
+	ElementName  string             `json:"name"`
+	Value        string             `json:"value"`
+	CallbackID   string             `json:"callback_id"`
+}
+
+// OpenDialog opens a dialog window where the triggerId originated from
+func (api *Client) OpenDialog(triggerId string, dialog Dialog) (err error) {
+	return api.OpenDialogContext(context.Background(), triggerId, dialog)
+}
+
+// OpenDialogContext opens a dialog window where the triggerId originated from with a custom context
+func (api *Client) OpenDialogContext(ctx context.Context, triggerId string, dialog Dialog) (err error) {
+	if triggerId == "" {
+		return errors.New("received empty parameters")
+	}
+
+	resp := DialogTrigger{
+		TriggerId:  triggerId,
+		Dialog:     dialog,
+	}
+	jsonResp, _ := json.Marshal(resp)
```

Comment:
i know its unlikely but lets not ignore marshalling errors.

Reply:
No problem, updating it now!

Reply:
Done!